scanner/src/main/kotlin/provenance/NestedProvenanceScanResult.kt

```diff
@@ -198,6 +168,53 @@ data class NestedProvenanceScanResult(
     }
 }
 
+/**
+ * Merge the nested [ScanResult]s into one [ScanResult] per used scanner, using the provenance from the [ScanResult]
+ * associated with the empty path as provenance. This maps the given [scanResultsByPath] to the format currently used
+ * by [OrtResult]. When merging multiple [ScanSummary]s the earliest start time will be used as the new start time, and
+ * the latest end time will be used as the end time. Because the [ScanSummary] does not contain the checksums of the
+ * individual files, no package verification code can be calculated.
+ */
+private fun mergeScanResults(scanResultsByPath: Map<String, List<ScanResult>>): List<ScanResult> {
```

Comment:
I would prefer `mergeScanResultsByScanner` as name as it would explain why the function returns a list of scan results.